### ISSUE TITLE
Enable native null assertions by default in Flutter web

### ISSUE THREAD

**Issue by srujzs** [dart-lang/sdk#44259]
Parent bug: #43780 

This flag needs to be turned on by default in both compilers. The work here involves not only enabling it by default, but providing a way for users to disable it to be unblocked. We'd like to be able to link the documentation to disable it as well in the markdown file below like we did for build_web_compilers. Ideally, we'd like to get this in before the stable release for Flutter web, so we can tackle any incoming issues around it. Currently, this flag will only throw errors if in sound mode on both web compilers.

More information on what this flag does and what it looks like in both compilers here:
https://github.com/dart-lang/sdk/blob/master/sdk/lib/html/doc/NATIVE_NULL_ASSERTIONS.md

**Comment by nshahan** [comment]
Just for clarification, do we want this enabled by default, in flutter web, in both sound and weak modes?

**Comment by srujzs** [comment]
It can be enabled by default in both, but in weak mode, it will do nothing. It's also possible that at some point in the future we might want to support opt-in libraries that use `dart:html`. In that case, maybe we'd rather not have it enabled by default in weak mode so that it doesn't trip Flutter code unless Flutter web decides to enable it then.

**Comment by srujzs** [comment]
Closing in favor of https://github.com/flutter/flutter/issues/71614.
